pkg/netceptor/tlsconfig.go

```diff
@@ -39,14 +41,39 @@ func decodeFingerprints(fingerprints []string) ([][]byte, error) {
 	return fingerprintBytes, nil
 }
 
+func checkCertificatesMatchNodeID(certbytes []byte, n *Netceptor, certName string, certPath string) error {
+	block, _ := pem.Decode(certbytes)
+
+	if block == nil {
+		return fmt.Errorf("failed to parse certfifcate PEM")
+	}
+
+	parsedCert, err := x509.ParseCertificate(block.Bytes)
+	if err != nil {
+		return err
+	}
+
+	found, receptorNames, err := utils.ParseReceptorNamesFromCert(parsedCert, n.nodeID)
+	if err != nil {
+		return err
+	}
+
+	if !found {
+		return fmt.Errorf("MainInstance.nodeID=%s not found in certificate name(s); names found=%s; cfg section=%s; server cert=%s", MainInstance.nodeID, fmt.Sprint(receptorNames), certName, certPath)
+	}
+
+	return nil
+}
+
 // tlsServerCfg stores the configuration options for a TLS server.
 type tlsServerCfg struct {
-	Name              string   `required:"true" description:"Name of this TLS server configuration"`
-	Cert              string   `required:"true" description:"Server certificate filename"`
-	Key               string   `required:"true" description:"Server private key filename"`
-	RequireClientCert bool     `description:"Require client certificates" default:"false"`
-	ClientCAs         string   `description:"Filename of CA bundle to verify client certs with"`
-	PinnedClientCert  []string `description:"Pinned fingerprint of required client certificate"`
+	Name                   string   `required:"true" description:"Name of this TLS server configuration"`
+	Cert                   string   `required:"true" description:"Server certificate filename"`
+	Key                    string   `required:"true" description:"Server private key filename"`
+	RequireClientCert      bool     `required:"false" description:"Require client certificates" default:"false"`
+	ClientCAs              string   `required:"false" description:"Filename of CA bundle to verify client certs with"`
+	PinnedClientCert       []string `required:"false" description:"Pinned fingerprint of required client certificate"`
+	SkipReceptorNamesCheck bool     `required:"false" description:"if enabled, validate cert using ReceptorNames OID in Certificate"`
```

Comment:
the description for this is tricky, I think we want to emphasize this boolean only controls the check at startup, not during the quic TLS handshake (where the receptor names will always be checked, regardless of how this boolean is set)

maybe something like,

`Skip verifying Receptor node ID in certificate at startup`

Reply:
done @fosterseth